DisCatSharp.Interactivity/Extensions/InteractionExtensions.cs

```diff
@@ -51,5 +53,72 @@ public static class InteractionExtensions
 		/// <param name="token">A custom cancellation token that can be cancelled at any point.</param>
 		public static Task SendPaginatedResponseAsync(this DiscordInteraction interaction, bool ephemeral, DiscordUser user, IEnumerable<Page> pages, PaginationButtons buttons = null, PaginationBehaviour? behaviour = default, ButtonPaginationBehavior? deletion = default, CancellationToken token = default)
 			=> MessageExtensions.GetInteractivity(interaction.Message).SendPaginatedResponseAsync(interaction, ephemeral, user, pages, buttons, behaviour, deletion, token);
+
+		/// <summary>
+		/// Sends multiple modals to the user with a prompt to open the next one.
+		/// </summary>
+		/// <param name="interaction">The interaction to create a response to.</param>
+		/// <param name="builder">The paginated modal builder.</param>
+		/// <param name="timeOutOverride">A custom timeout. (Default: 15 minutes)</param>
+		/// <returns>A read-only dictionary with the customid of the components as the key.</returns>
+		/// <exception cref="ArgumentException">Is thrown when no modals are defined.</exception>
+		/// <exception cref="InvalidOperationException">Is thrown when interactivity is not enabled for the client/shard.</exception>
+		public static async Task<PaginatedModalResponse> CreatePaginatedModalResponseAsync(this DiscordInteraction interaction, DiscordInteractionPaginatedModalBuilder builder, TimeSpan? timeOutOverride = null)
+		{
+			if (builder.Modals is null || builder.Modals.Count is 0)
+				throw new ArgumentException("You have to set at least one page");
+
+			var client = (DiscordClient)interaction.Discord;
+			var interactivity = client.GetInteractivity() ?? throw new InvalidOperationException($"Interactivity is not enabled for this {(client.IsShard ? "shard" : "client")}.");
+
+			timeOutOverride = timeOutOverride is null ? TimeSpan.FromMinutes(15) : timeOutOverride;
```

Comment:
Why declare it nullable when logically it's non-nullable with a default value of 15 minutes?
If sending a request with a null timeout override works we should probably just do that because it allows us to ignore changes to the default value on Discord's backend side.

Reply:
I don't quite understand what you mean. You can't set a default value of 15 minutes in the overloads because that wouldn't be constant.
![image](https://user-images.githubusercontent.com/56395159/168139932-3a75ec8b-c009-47c0-be40-fbd005195f94.png)
![image](https://user-images.githubusercontent.com/56395159/168139948-bfafc398-1ae1-4d3f-8294-e2584404c827.png)

Reply:
After taking a second look.. At least use `??=` ^^

Reply:
Fixed